native/src/main/java/io/ballerina/stdlib/rabbitmq/MessageDispatcher.java

```diff
@@ -188,8 +199,11 @@ private Object[] getResourceParameters(byte[] message, Envelope envelope, AMQP.B
                             returnErrorValue("Invalid remote function signature");
                         }
                         messageExists = true;
-                        arguments[index++] = createAndPopulateMessageRecord(message, envelope,
+                        Object record = createAndPopulateMessageRecord(message, envelope,
                                 properties, referredType);
+                        validateConstraints(record, getElementTypeDescFromArrayTypeDesc(ValueCreator
+                                .createTypedescValue(parameter.type)), constraintValidation);
+                        arguments[index++] = record;
                         arguments[index++] = true;
```

Comment:
As a suggestion, we can move this `arguments[index++] = true;` line to the end of the switch and do it once.